portal/views.py

```diff
@@ -61,23 +61,27 @@ def get_public_apps(request, app_id=0, collection=None, order_by="updated_on", o
         if "tf_add" not in request.GET and "tf_remove" not in request.GET:
             request.session["app_tags"] = {}
 
+    # Select published apps
     published_apps = []
+    # because shiny appears twice we have to ensure uniqueness
+    seen_app_ids = set()
+
+    def process_queryset(queryset):
+        for app in queryset:
+            if app.id not in seen_app_ids:
+                published_apps.append(app)
+                seen_app_ids.add(app.id)
```

Comment:
```suggestion
    # Select published apps
    published_apps = []
    # because shiny appears twice we have to ensure uniqueness
    seen_app_ids = set()

    def get_unique_apps(queryset, app_ids_to_exclude):
        """Get from queryset app orm models, that are not present in ``seen_app_ids``"""
        unique_app_ids_ = set()
        unique_apps_ = []
        for app in queryset:
            if app.id not in app_ids_to_exclude and app_id not in unique_app_ids_:
                unique_app_ids_.add(app.id)
                unique_apps_.append(app)
        return unique_apps_, unique_app_ids_
```

Not insisting on the `app_ids_to_exclude` name, could be something else)